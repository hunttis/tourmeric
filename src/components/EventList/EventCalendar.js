import React, { Component } from 'react';
import { isLoaded, isEmpty } from 'react-redux-firebase';
import _ from 'lodash';
import { Translate } from 'react-localize-redux';
import PropTypes from 'prop-types';
import moment from 'moment/min/moment-with-locales';
import EventCard from './EventCard-container';
import { CalendarMonth } from './CalendarMonth';

export default class EventCalendar extends Component {

  constructor(props) {
    super(props);

    const favoriteCategories = _.get(props, 'profile.favoriteCategories', '');
    const defaultFilter = favoriteCategories.split(' ');
    this.state = { categoryFilter: _.compact(defaultFilter), viewedDate: moment() };
  }

  toggleFilter(categoryId) {
    const { categoryFilter } = this.state;
    if (_.includes(categoryFilter, categoryId)) {
      const newFilter = categoryFilter.slice();
      newFilter.splice(newFilter.indexOf(categoryId), 1);
      this.setState({ categoryFilter: newFilter });
    } else {
      const newFilter = categoryFilter.slice();
      newFilter.push(categoryId);
      this.setState({ categoryFilter: newFilter });
    }
  }

  runEventFilters(events) {
    const { categoryFilter } = this.state;

    const publishedEvents = events.filter(event => event.value.published);
    if (_.isEmpty(categoryFilter)) {
      return publishedEvents;
    }

    const publishedAndFilteredEvents = publishedEvents.filter(event => _.includes(this.state.categoryFilter, event.value.category));
    return publishedAndFilteredEvents;
  }

  modalItem(translationKey, content) {
    if (content) {
      return (
        <div className="column is-12">
          <div className="title"><Translate id={translationKey} /></div>
          <p>
            {content}
          </p>
        </div>
      );
    }
    return <div />;
  }

  forwardMonth() {
    this.setState(prevState => ({ viewedDate: prevState.viewedDate.add(1, 'month') }));
  }

  backMonth() {
    this.setState(prevState => ({ viewedDate: prevState.viewedDate.subtract(1, 'month') }));
  }

  clickDay(day) {
    this.props.history.push(`/events/${day.dayLink}`);
  }

  backToCalendar() {
    this.props.history.push('/events');
  }

  render() {
    const {
      events, participations, profile, categories, settings, uploadedCategoryLogos, activeLanguage, location,
    } = this.props;

    const { viewedDate } = this.state;

    if (!isLoaded(participations) || !isLoaded(profile) || !isLoaded(events) || !isLoaded(settings) || !isLoaded(uploadedCategoryLogos) || !isLoaded(categories)) {
      return <div className="is-loading"><Translate id="loading" /></div>;
    } if (isLoaded(events) && isEmpty(events)) {
      return <div><Translate id="noevents" /></div>;
    }
    const publishedEvents = this.runEventFilters(events);

    const categoryNames = this.state.categoryFilter.map(category => categories[category].name).join(', ');

    const dayCount = viewedDate.daysInMonth();
    const days = [];

    moment.locale(activeLanguage);

    for (let i = 1; i <= dayCount; i += 1) {
      const dayString = `${_.padStart(i, 2, '0')}-${viewedDate.format('MM-YYYY')}`;
      const day = moment(dayString, 'DD-MM-YYYY');
      const dayStringInEventFormat = moment(dayString, 'DD-MM-YYYY').format('YYYY-MM-DD');
      const eventsForDay = publishedEvents.filter(eventEntry => eventEntry.value.date === dayStringInEventFormat);
      days.push({ day: day.format('DD'), dayOfWeek: day.format('d'), dayName: day.format('dddd'), dayString: day.format('DD-MMMM-YYYY'), dayLink: day.format('DD/MM/YYYY'), eventsForDay });
    }

    const emptyDays = (days[0].dayOfWeek % 7) - 1;
    const calendar = [];
    for (let i = 0; i < emptyDays; i += 1) {
      calendar.push({ empty: true });
    }
    _.forEach(days, day => calendar.push(day));
    const chunkedCalendar = _.chunk(calendar, 7);
    while (chunkedCalendar[chunkedCalendar.length - 1].length < 7) {
      chunkedCalendar[chunkedCalendar.length - 1].push({ empty: true });
    }

    let dayModalOpen = false;
    let eventsForDay = {};

    if (location.pathname.split('/').length > 2) {
      eventsForDay = _.get(_.find(calendar, { dayLink: location.pathname.substring('/events/'.length) }), 'eventsForDay', {});
      dayModalOpen = true;
    }

    return (
      <section className="section">
        {dayModalOpen &&
          <div className="modal is-active">
            <div className="modal-background" onClick={() => this.backToCalendar()} />
            <div className="modal-content box">
              {_.isEmpty(eventsForDay) && <Translate id="noeventsforthisday" />}
              {!_.isEmpty(eventsForDay) &&
              <div>
                {eventsForDay.map((eventEntry, index) => {
                  const eventId = eventEntry.key;
                  return <EventCard key={`events-for-day-${index}`} eventId={eventId} />;
                })}
              </div>
            }

            </div>
            <button className="modal-close is-large" aria-label="close" onClick={() => this.backToCalendar()} />
          </div>
        }
        <div className="container">

          <h1 className="title"><Translate id="nextevents" /></h1>

          <div className="columns is-multiline">

            <div className="column is-12">
              <Translate id="filtereventsbychoosingcategories" />
              <Translate>
                { translate => (
                  <div>
                    {translate('currentlyshowing')}
                    {_.isEmpty(this.state.categoryFilter) ? translate('all') : translate('only')} <span className="has-text-success">{categoryNames}</span> {translate('events')}
                  </div>
                )}
              </Translate>

            </div>

            <div className="column is-12 is-mobile has-text-centered">
              {isLoaded(categories) && Object.entries(categories).map((categoryEntry) => {
                const activeStatus = _.includes(this.state.categoryFilter, categoryEntry[0]) ? 'is-primary' : '';
                const buttonClass = `button is-grouped image-square ${activeStatus} `;

                const category = categories[categoryEntry[0]];

                return (
                  <div key={`categoryfilter-${categoryEntry[0]}`} className="is-one-quarter-desktop is-one-quarter-tablet is-one-third-mobile has-text-centered is-inline-flex">
                    <button
                      className={buttonClass}
                      onClick={() => this.toggleFilter(categoryEntry[0])}
                    >
                      <img className="image is-48x48" src={category.image} alt="" />
                    </button>
                  </div>
                );
              })
              }
            </div>

            <div className="column is-6">
              <h1 className="title">{_.capitalize(viewedDate.format('MMMM'))}</h1>
            </div>
            <div className="column is-6">
              <div className="buttons has-addons is-pulled-right">
                <button className="button" onClick={() => { this.backMonth(); }}><Translate id="previousmonth" /></button>
                <button className="button" onClick={() => { this.forwardMonth(); }}><Translate id="nextmonth" /></button>
              </div>
            </div>

            <CalendarMonth
              chunkedCalendar={chunkedCalendar}
              categories={categories}
              clickDay={day => this.clickDay(day)}
            />
          </div>
        </div>
      </section>
    );
  }
}


EventCalendar.propTypes = {
  events: PropTypes.array,
  participations: PropTypes.object,
  profile: PropTypes.object,
  categories: PropTypes.object,
  settings: PropTypes.object,
  uploadedCategoryLogos: PropTypes.object,
  activeLanguage: PropTypes.string,
  location: PropTypes.object,
  history: PropTypes.object,
};
