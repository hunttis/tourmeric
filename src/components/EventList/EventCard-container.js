import { connect } from 'react-redux';
import { compose } from 'redux';
import EventCard from './EventCard';

export default compose(
  connect(state => ({
    events: state.firebase.ordered.events,
    participations: state.firebase.data.participations,
    categories: state.firebase.data.categories,
    userid: state.firebase.auth.uid,
    settings: state.firebase.data.settings,
    languages: state.locale.languages,
  })),
  connect(({ firebase: { profile } }) => ({ profile })),
)(EventCard);
