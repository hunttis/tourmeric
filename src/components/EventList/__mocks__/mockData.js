export const mockEventId = 'mockEventId1';

export const mockUserId = 'mockUserId';

export const mockProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{ displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890' }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
};

export const mockUnorderedEvents = {
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents = [
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png' },
  };

export const mockParticipations = { mockEventId1: { mockUserId: { comment: 'Jee',
  date: '2018-06-23T09:40:26.992Z',
  firstName: 'Test',
  lastName: 'Person',
  userId: 'mockUserId' } } };

export const mockCategories = { category1: { abbreviation: 'WH40K',
  logo: '1527227370219137123',
  name: 'Warhammer 40,000',
  type: 'Miniatures' },
category2: { abbreviation: 'MtG',
  logo: '152722965771313789',
  name: 'Magic: The Gathering',
  type: 'TCG' },
'-LAlvCmIWNzYKpa7-7HI': { abbreviation: 'Lautapelit',
  logo: '152719272536668394',
  name: 'Lautapelit / Board games' },
'-LCmESny8m9JVEFQ_qOt': { abbreviation: 'PKM',
  logo: '152722747904159438',
  name: 'Pok√©mon',
  type: 'TCG' },
'-LDImTAGxkzrorL0tQdi': { abbreviation: 'VG',
  logo: '152719372155055672',
  name: 'Videogames' },
'-LDIw-1LFgiM0EJZGfus': { abbreviation: 'Necromunda',
  logo: '15271953483568546',
  name: 'Necromunda',
  type: 'Miniatures' } };
