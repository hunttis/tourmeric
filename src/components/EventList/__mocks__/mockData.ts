import { FirebaseProfile, Participation, User, ProviderData, Location } from '~/models/ReduxState';
import { TourmericEvent } from '~/models/Events';
import { Settings } from '~/models/Settings';
import { UploadedFile, Category } from '~/models/Category';

export const mockEventId = 'mockEvent1';

export const mockUserId = 'mockUserId';

export const mockLocation: Location = {
  pathname: '/events/2018/10',
  search: 'search',
  hash: 'hash',
  key: 'key',
};

export const mockProviderData: ProviderData = {
  displayName: 'MockName',
  email: 'mock@userton.com',
  photoURL: 'http://url',
  providerId: 'facebook',
  uid: '1234567890',
};

export const mockUser: User = {
  avatarUrl: 'http://www.google.fi/google.png',
  displayName: 'Mock Userton',
  firstName: 'Mock',
  lastName: 'Userton',
  providerData: mockProviderData,
  email: 'mock@userton.com',
  username: 'Mockie',
  dciNumber: '123456',
  role: 'user',
  active: true,
};

export const mockProfile: FirebaseProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{
    displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890',
  }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent: TourmericEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
  eventType: 'single',
};

export const mockUnorderedEvents: {[key: string]: TourmericEvent} = {
  mockEvent1: mockSingleEvent,
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents: {key: string, value: TourmericEvent}[] = [
  { key: 'mockEvent1',
    value: mockSingleEvent },
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings: Settings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos: {[key: string]: UploadedFile} =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png', name: 'image' },
  };

export const mockParticipations: {[key: string]: Participation} = {
  mockEventId1: {
    mockUserId: {
      comment: 'Jee',
      date: '2018-06-23T09:40:26.992Z',
      firstName: 'Test',
      lastName: 'Person',
      userId: 'mockUserId',
    },
  },
};

export const mockCategories: {[key: string]: Category} = {
  category1: {
    abbreviation: 'WH40K',
    logo: '1527227370219137123',
    name: 'Warhammer 40,000',
    type: 'Miniatures',
  },
  category2: {
    abbreviation: 'MtG',
    logo: '152722965771313789',
    name: 'Magic: The Gathering',
    type: 'TCG',
  },
  '-LAlvCmIWNzYKpa7-7HI': {
    abbreviation: 'Lautapelit',
    logo: '152719272536668394',
    name: 'Lautapelit / Board games',
    type: 'Boardgame',
  },
  '-LCmESny8m9JVEFQ_qOt': {
    abbreviation: 'PKM',
    logo: '152722747904159438',
    name: 'Pok√©mon',
    type: 'TCG',
  },
  '-LDImTAGxkzrorL0tQdi': {
    abbreviation: 'VG',
    logo: '152719372155055672',
    name: 'Videogames',
    type: 'Videogames',
  },
  '-LDIw-1LFgiM0EJZGfus': {
    abbreviation: 'Necromunda',
    logo: '15271953483568546',
    name: 'Necromunda',
    type: 'Miniatures',
  },
};
