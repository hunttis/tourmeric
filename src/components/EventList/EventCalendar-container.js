import { connect } from 'react-redux';
import { compose } from 'redux';
import { getActiveLanguage } from 'react-localize-redux';
import EventCalendar from './EventCalendar';

export default compose(
  connect(state => ({
    events: state.firebase.ordered.events,
    participations: state.firebase.data.participations,
    categories: state.firebase.data.categories,
    userid: state.firebase.auth.uid,
    settings: state.firebase.data.settings,
    languages: state.locale.languages,
    uploadedCategoryLogos: state.firebase.data.uploadedCategoryLogos,
    activeLanguage: getActiveLanguage(state.locale).code,
  })),
  connect(({ firebase: { profile } }) => ({ profile })),
)(EventCalendar);
