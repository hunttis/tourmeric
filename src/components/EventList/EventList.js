import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import EventCalendar from './EventCalendar-container';
import EventListOld from './EventListOld-container';

export default class EventList extends Component {

  foo() {}

  render() {
    const { profile } = this.props;

    if (_.get(profile, 'betaparticipation.eventsPreference', '') === 'calendar') {
      return <EventCalendar />;
    }
    return <EventListOld />;
  }
}

EventList.propTypes = {
  profile: PropTypes.object,
};
