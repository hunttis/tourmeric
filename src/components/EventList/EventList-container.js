import { connect } from 'react-redux';
import { compose } from 'redux';
import { firebaseConnect } from 'react-redux-firebase';
import { getActiveLanguage } from 'react-localize-redux';
import EventList from './EventList';

export default compose(
  firebaseConnect([
    { path: '/categories' },
    { path: '/participations' },
    { path: '/uploadedCategoryLogos' },
  ]),
  connect(state => ({
    events: state.firebase.ordered.events,
    participations: state.firebase.data.participations,
    categories: state.firebase.data.categories,
    userid: state.firebase.auth.uid,
    settings: state.firebase.data.settings,
    languages: state.locale.languages,
    uploadedCategoryLogos: state.firebase.data.uploadedCategoryLogos,
    activeLanguage: getActiveLanguage(state.locale).code,
  })),
  connect(({ firebase: { profile } }) => ({ profile })),
)(EventList);
