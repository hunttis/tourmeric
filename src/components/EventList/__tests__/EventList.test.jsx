import React from 'react';
import { shallow } from 'enzyme';
import EventList from '../EventList';
import { mockEvents, mockParticipations, mockProfile, mockCategories, mockSettings, mockUploadedLogos } from '../__mocks__/mockData';

describe('EventList tests', () => {

  it('Compares EventList to snapshot', () => {

    const eventList = shallow(<EventList
      events={mockEvents}
      participations={mockParticipations}
      profile={mockProfile}
      categories={mockCategories}
      settings={mockSettings}
      uploadedCategoryLogos={mockUploadedLogos}
    />);

    expect(eventList).toMatchSnapshot();
  });
});
