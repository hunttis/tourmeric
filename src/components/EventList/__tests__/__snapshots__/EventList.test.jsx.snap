// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList tests Compares EventList to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventList
    categories={
      Object {
        "-LAlvCmIWNzYKpa7-7HI": Object {
          "abbreviation": "Lautapelit",
          "logo": "152719272536668394",
          "name": "Lautapelit / Board games",
        },
        "-LCmESny8m9JVEFQ_qOt": Object {
          "abbreviation": "PKM",
          "logo": "152722747904159438",
          "name": "PokÃ©mon",
          "type": "TCG",
        },
        "-LDImTAGxkzrorL0tQdi": Object {
          "abbreviation": "VG",
          "logo": "152719372155055672",
          "name": "Videogames",
        },
        "-LDIw-1LFgiM0EJZGfus": Object {
          "abbreviation": "Necromunda",
          "logo": "15271953483568546",
          "name": "Necromunda",
          "type": "Miniatures",
        },
        "category1": Object {
          "abbreviation": "WH40K",
          "logo": "1527227370219137123",
          "name": "Warhammer 40,000",
          "type": "Miniatures",
        },
        "category2": Object {
          "abbreviation": "MtG",
          "logo": "152722965771313789",
          "name": "Magic: The Gathering",
          "type": "TCG",
        },
      }
    }
    events={
      Array [
        Object {
          "key": "mockEvent2",
          "value": Object {
            "category": "category2",
            "createDate": "2018-05-20T16:46:32.009Z",
            "date": "2018-05-20",
            "entryFee": "6",
            "format": "Standard",
            "name": "Tapahtuma",
            "notes": "FIJoijgasdsdgasdggagads",
            "playerSlots": "14",
            "prizes": "Foooooooooooooooo",
            "rulesLevel": "Competitive",
            "time": "10:00",
          },
        },
        Object {
          "key": "mockEvent3",
          "value": Object {
            "category": "category2",
            "createDate": "2018-05-20T16:04:29.595Z",
            "date": "2018-05-21",
            "entryFee": "5",
            "format": "Modern",
            "link": "https://magic.wizards.com/en/game-info/gameplay/formats/modern",
            "name": "Monday Modern",
            "notes": "Bring a friend!",
            "playerSlots": "31",
            "prizes": "Three (3) OMG credits per player into the prizepool.",
            "published": true,
            "rulesLevel": "Regular",
            "time": "19:00",
          },
        },
        Object {
          "key": "mockEvent4",
          "value": Object {
            "category": "-LDIw-1LFgiM0EJZGfus",
            "createDate": "2018-05-24T20:58:06.938Z",
            "date": "2018-05-24",
            "entryFee": "5",
            "format": "League",
            "link": "jokulinkki.fi",
            "name": "Necromunda League",
            "notes": "Shame! Shame! Shame! *DING DING DING*",
            "playerSlots": "12",
            "prizes": "I am Loot.",
            "published": true,
            "rulesLevel": "Casual",
            "time": "17:00",
          },
        },
        Object {
          "key": "mockEvent5",
          "value": Object {
            "category": "-LAlvCmIWNzYKpa7-7HI",
            "createDate": "2018-05-24T17:01:53.909Z",
            "date": "2018-05-26",
            "entryFee": "0",
            "format": "Board game",
            "link": "https://geekandsundry.com/international-tabletop-day-april-28-2018/",
            "name": "Lautapeli Lauantai",
            "notes": "Bring a friend and get a surprise gift!",
            "playerSlots": "30",
            "prizes": "A Supi's surprise prize for all participants!",
            "published": true,
            "rulesLevel": "Casual",
            "time": "00:00",
          },
        },
        Object {
          "key": "mockEventId1",
          "value": Object {
            "category": "category1",
            "createDate": "2018-05-24T17:00:46.500Z",
            "date": "2018-09-24",
            "entryFee": "10",
            "format": "Miniatures",
            "link": "http://www.google.fi",
            "name": "Warhammer thing",
            "notes": "Bring your war face!",
            "playerSlots": "10",
            "prizes": "Glory!",
            "published": true,
            "rulesLevel": "Casual",
            "time": "Kill everything",
          },
        },
      ]
    }
    participations={
      Object {
        "mockEventId1": Object {
          "mockUserId": Object {
            "comment": "Jee",
            "date": "2018-06-23T09:40:26.992Z",
            "firstName": "Test",
            "lastName": "Person",
            "userId": "mockUserId",
          },
        },
      }
    }
    profile={
      Object {
        "avatarUrl": "https://foo.com/picture",
        "dciNumber": "123512351112",
        "displayName": "Test Person",
        "email": "fake@fake.com",
        "firstName": "Test",
        "isEmpty": false,
        "isLoaded": true,
        "lastName": "Person",
        "providerData": Array [
          Object {
            "displayName": "Test Person",
            "email": "fake@fake.com",
            "photoURL": "https://foo.com/picture",
            "providerId": "facebook.com",
            "uid": "1234567890",
          },
        ],
        "role": "admin",
      }
    }
    settings={
      Object {
        "activeLogo": "http://foo.com/image.png",
        "dateFormat": "DD.MM.YYYY",
        "pageSubtitle": "",
        "pageTitle": "",
        "subtitleTextColor": "#000",
        "theme": "darkly",
        "titleBarAngle": "150",
        "titleBarColor": "#000",
        "titleBarColor2": "#ec1c22",
        "titleBarPercentage": "50",
        "titleTextColor": "",
      }
    }
    uploadedCategoryLogos={
      Object {
        "152719272536668394": Object {
          "downloadURL": "http://foo.com/image.png",
        },
        "152719372155055672": Object {
          "downloadURL": "http://foo.com/image.png",
        },
        "15271953483568546": Object {
          "downloadURL": "http://foo.com/image.png",
        },
        "1527227370219137123": Object {
          "downloadURL": "http://foo.com/image.png",
        },
        "152722747904159438": Object {
          "downloadURL": "http://foo.com/image.png",
        },
        "152722965771313789": Object {
          "downloadURL": "http://foo.com/image.png",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
